pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -362,6 +367,115 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance) {
 	}(l, vmi)
 }
 
+func getVMIEphemeralDisksTotalSize() *resource.Quantity {
+	var baseDir = "/var/run/kubevirt-ephemeral-disks/"
+	totalSize := int64(0)
+	err := filepath.Walk(baseDir, func(path string, f os.FileInfo, err error) error {
+		if !f.IsDir() {
+			totalSize += f.Size()
+		}
+		return err
+	})
+	if err != nil {
+		log.Log.Reason(err).Warning("failed to get VMI ephemeral disks size")
+		return &resource.Quantity{Format: resource.BinarySI}
+	}
+
+	return resource.NewScaledQuantity(totalSize, 0)
+}
+
+func getVMIMigrationDataSize(vmi *v1.VirtualMachineInstance) int64 {
+	var memory resource.Quantity
+
+	// Take memory from the requested memory
+	if v, ok := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]; ok {
+		memory = v
+	}
+	// In case that guest memory is explicitly set, override it
+	if vmi.Spec.Domain.Memory != nil && vmi.Spec.Domain.Memory.Guest != nil {
+		memory = *vmi.Spec.Domain.Memory.Guest
+	}
+
+	//get total data Size
+	if vmi.Status.MigrationMethod == v1.BlockMigration {
+		disksSize := getVMIEphemeralDisksTotalSize()
+		memory.Add(*disksSize)
+	}
+	return memory.ScaledValue(resource.Giga)
+}
+
+func liveMigrationMonitor(vmi *v1.VirtualMachineInstance, dom cli.VirDomain, l *LibvirtDomainManager) {
+	logger := log.Log.Object(vmi)
+	start := time.Now().UTC().Unix()
+	lastProgressUpdate := start
+	progressWatermark := int64(0)
+	// TODO:(vladikr) move to configMap
+	progressTimeout := int64(150)
```

Comment:
Maybe it makes sense to have this progressTimeout as an option in the Migration API.